### DISCUSSION TITLE
v6.0.0

### DISCUSSION THREAD

**Discussion by Ghost-chu** [PBH-BTN/PeerBanHelper#332] [公告]
> [!NOTE]
> 这是一个**候选发布版本**，并非正式版，请不要用于生产环境。**升级前请做好数据备份**。如果遇到错误，欢迎[报告错误](https://github.com/PBH-BTN/PeerBanHelper/issues)。

> [!NOTE]
> v5 升级到 v6 的用户：建议您删除 data/scripts 文件夹，以便 PeerBanHelper 释放新的检测用户脚本，以对抗最近新出现的无法使用 IP 规则解决的威胁

> [!WARNING]
> v6 与 v5 版本的 Banlist 不兼容，升级将会清空当前封禁列表。

### 日流量 & 趋势图图表

日流量：可展示每日 PBH 统计的上传与下载流量数据（不支持 Transmission）（可能与真实数据和下载器统计数据有较大出入，仅供参考）  
趋势：可展示指定时间段内的正常 Peers 与被封禁的 Peers 占比数据。

| 日流量 | 趋势 |
| ------ | ------- |
| ![日流量](https://github.com/user-attachments/assets/7d2670f5-cba3-419b-8250-b78d0034b565) | ![趋势](https://github.com/user-attachments/assets/3a709373-8209-4b3e-bb8a-0fe55df4aba1) |

### 位置与运营商

> [!NOTE]
> 使用此功能需要先[配置 GeoIP 库](https://github.com/PBH-BTN/PeerBanHelper/wiki/%E7%94%B3%E8%AF%B7-Maxmind-GeoLite2-%E8%AE%B8%E5%8F%AF%E8%AF%81)

显示 IP 的有关 GeoIP 数据，点击折叠菜单的切换开关，可在全量数据与仅封禁数据之间切换。

| ISP/AS | 省 | 城市 | 国家/地区 |
| ------   | ---- | ----- | -------|
| ![isp/as](https://github.com/user-attachments/assets/d98ea741-fafd-4d48-ac47-257cae9322d4) | ![省](https://github.com/user-attachments/assets/c82db42d-e2b0-4f10-909e-d67fa52d4279) | ![城市](https://github.com/user-attachments/assets/805b339b-677c-4760-8941-015d6ecc11b6) | ![国家地区](https://github.com/user-attachments/assets/1fde337b-58c2-424e-8a69-19c0b902256f) |

### Web 规则管理

新增 IP、端口、ASN、国家/地区、城市 规则管理页面，管理相关规则无需编辑配置文件
![image](https://github.com/user-attachments/assets/e5008985-dd63-4cd2-a780-4eb198cbaeae)

### 快速威胁分析

封禁列表的解除封禁按钮旁边现在新增一个威胁分析查询按钮，可在[微步威胁情报社区](https://x.threatbook.com/)（使用简体中文时）和 [Censys Search](https://search.censys.io/) （使用 English 时）快速跳转查询指定 IP 地址的威胁和特征信息。

![image](https://github.com/user-attachments/assets/514033fa-78aa-4946-a911-f10f294e2221)

![image](https://github.com/user-attachments/assets/c6baf944-1036-4985-842f-551b007bd948)

### Windows 平台能耗优化

在支持的 Windows 平台设备上，现在为 PeerBanHelper 启用 EcoQoS API 效能模式，优化功耗。这可以有效避免 PBH 执行规则检查时，引发的频繁 CPU 短暂睿频导致的能源消耗。  
启用成功时，除了控制台会出现成功提示，GUI 窗口标题将新增显示 `[🍃EcoQoS]` 的标识。

![d77ac4097d0dbfead1c713b6c15c7404](https://github.com/user-attachments/assets/cd437930-751b-4356-8ea0-8941ef3742a4)

### 其它更改

* IP 过滤器现在支持按照城市名称过滤（GeoCN 将按照完整名称过滤： 省份/自治区 城市/直辖市 区/县）
* 订阅规则现在支持 eMule/DAT 规则（将转换为 CIDR）
* 封禁排行现在支持翻页
* 下载器封禁列表提供端点新增 RawIP 类型
* GUI 现在适配系统亮色/暗色模式，系统模式切换时，PBH 将跟随切换其 GUI 主题
  * 已知在切换颜色模式时，可能出现切不回去的 BUG，暂时无修复思路
* GUI 新增 “调试” 菜单，可转储内存快照和打印线程堆栈

## 配置文件更改

* `Gopeed dev`, `Rain 0.0.0` 在升级后将加入默认屏蔽名单
* all-in-one 规则订阅源将切换为 CloudFlare Pages。这改善了在不少地区的可访问性，但仍建议根据 https://github.com/PBH-BTN/BTN-Collected-Rules 配置最适合您的规则源

## 错误修复

* 添加的 Transmission 下载器可能会错误的重新启动已被用户手动暂停的种子，现在原本已处于暂停状态的种子将被跳过重启环节
* 为部分配置项添加默认内容，避免被 SnakeYAML 吞掉
* 修复订阅日志排序异常的问题
* 修复下载器配置 `non-proxy-hosts` 错误的问题，现在错误的配置字段将被自动迁移到正确的配置字段，默认值将被修改为 `localhost|127.*|192.168.*|10.*|172.16.*|172.17.*|172.18.*|172.19.*|172.20.*|172.21.*|172.22.*|172.23.*|172.24.*|172.25.*|172.26.*|172.27.*|172.28.*|172.29.*|172.30.*|172.31.*`
* 在发现 Peer 重复封禁时，PBH 将尝试重新应用全量封禁列表到下载器
* 修复封禁日志自动清理功能不起效果的问题
* 修复针对 `uTorrent Mac` 和 `Aria2Explorer` 的伪装检测误判问题
* 修复错误 Etag 头导致版本升级后白屏的问题
* 修复 BTN 上报在出现异常后，后续计划任务被停止的问题
* 修复下载器名称包含 `.` 后，配置文件损坏的问题，**建议您在任何输入名称的文本框中避免输入 `.` 符号**
* 修复 TopN 封禁排行榜页面不显示任何数据的问题
* 修复 ActiveMonitoring 统计数据不正确的问题
* 修复 BTN 网络模块重新配置时没有取消过往调度器导致频繁重复上报的问题
* 修复 BTN 协议版本号检查逻辑错误
* 现在 PBH 在检测到尝试封禁已处于封禁名单的 IP 时，将强制全量更新封禁列表到下载器，修复封禁列表失去同步的问题
* 修复进度检查器对 IPV6 地址处理错误导致误封禁的问题，感谢 @paulzzh

## 即将到来的新功能&未来计划

以下功能正在进行中或已计划，目前我们已完成基础架构编写，但仍需一段时日打磨、调校和 Debug，它们会在未来版本中推出：

* [进行中]配置文件重载
* [已完成，但API接口需要更改]WebUI 实时日志控制台
  * 提供网页控制台接口，可以查看控制台中打印的日志信息，不再需要查看 GUI/Docker Logs
* [已完成，等待前端适配]种子列表
  * 由 PBH 在检查过程中保存的种子数据列表，在误删种子后，提供一个后悔药
  * 当然更重要的是为下面的功能服务
* [已完成，等待前端适配]种子详细信息查询
  * 可查询以下数据
    * 种子名称
    * 种子大小
    * 种子 InfoHash
    * 连接过此种子的 Peers 数量
    * 连接过此种子的 Peers 详细信息
    * 在此种子上被封禁的 Peers 数量
    * 在此种子上被封禁的 Peers 详细信息
* [已完成，等待前端适配]Peer 详细信息查询
  * 可查询以下数据
    * IP 地址
    * 首次连接时间
    * 最后一次连接时间
    * 封禁计数
    * 种子访问会话数
    * 历史上传量
    * 历史下载量
    * GeoIP 信息
    * 访问过的所有种子列表，以及历史会话详细数据，包括
      * 历史会话 Torrent 信息
      * 历史会话下载器名称
      * 历史会话 PeerId
      * 历史会话 ClientName
      * 历史会话上传量下载量
      * 历史会话 uTP Flags
      * 历史会话开始时间
      * 历史会话结束时间

## 正式推出 PeerBanHelper+ 计划

> PBH-BTN 社区的所有项目都由我们的社区志愿者更新和维护。我们的志愿者每周会花费数十个小时来维护我们的社区项目并更新反吸血规则。尽管志愿者们都秉承无私奉献的精神，不求回报的参与其中。但除了提供精神上的支持以外，物质上的支持也同样重要
> 购买此许可证的费用将用于建设和维护社区基础设施，并分配给志愿者们作为物质上的奖励

我们将部分在开发过程中逻辑较复杂、耗时较久、技术难度较高的功能列入 PBH+ 计划。请不要担心，重要和基础功能永远对所有人可用。

您可以在这里查看 PBH+ 计划的定价和描述：[爱发电电铺](https://afdian.com/a/Ghost_chu?tab=shop)。

我们建议您仅在确实会用到 PBH+ 的功能时再购买 PBH+ 许可证，切勿掉入消费陷阱，造成不必要的额外开销。

在 v6 版本中，以下功能为 PBH+ 功能：

* [WebUI][PBH+] 新增趋势、日流量、位置与运营商三图表

## Docker 用户

DockerHub: `ghostchu/peerbanhelper:v6.0.0`   
阿里云国内镜像加速: `registry.cn-hangzhou.aliyuncs.com/ghostchu/peerbanhelper:v6.0.0`

<hr /><em>This discussion was created from the release <a href='https://github.com/PBH-BTN/PeerBanHelper/releases/tag/v6.0.0'>v6.0.0</a>.</em>

**Comment by GalaxySnail** [comment]
请问是否有 v5 到 v6 的封禁列表手动迁移步骤？另外在 v6 发布之后，v5 是否还会有错误修复更新？

**Reply by Ghost-chu** [reply]
> 请问是否有 v5 到 v6 的封禁列表手动迁移步骤？

您可以自行处理数据库中的 JSON 的差异。但我们不提供此类支持。

> 另外在 v6 发布之后，v5 是否还会有错误修复更新？

PBH 不会维护过往版本，建议保持更新。

**Comment by Einsfier** [comment]
非常棒的社区项目，已赞助Plus，建议打开Github sponsor
